File: hw/cañarte/u3/HW28SingletonAndAbstractFactory/Java/GUIFactory/src/main/java/ec/edu/espe/guifactory/controller/GUIFactory.java

package ec.edu.espe.guifactory.controller;

/**
 *
 * @author Saray Canarte, Code Creators, DCCO-ESPE
 */
public abstract class GUIFactory {
    public static GUIFactory getFactory() {
        int sys = readFromConfigFile("OS_TYPE");
        if (sys == 0)
            return new WinFactory();
        else
            return new LinuxFactory();
    }

    
    public abstract Button createButton();
    public abstract Menu createMenu();

   
    private static int readFromConfigFile(String key) {
        if (key.equals("OS_TYPE")) {
            // Assuming 0 represents Windows and 1 represents Linux
            return 1; 
        }
        return -1;
    }
}

File: hw/Rodriguez/u3/hw28/hw28Python/AbstractFactory/controller/GUIFactory.py
import sys
import os

sys.path.append(os.path.abspath(os.path.join(os.path.dirname(__file__), '..')))
from controller.Button import Button
from controller.Menu import Menu
from controller.WinFactory import WinFactory
from controller.LinuxFactory import LinuxFactory
from abc import ABC, abstractmethod


class GUIFactory(ABC):
    @staticmethod
    def get_factory():
        sys = GUIFactory.read_from_config_file("OS_TYPE")
        if sys == 0:
            return WinFactory()
        else:
            return LinuxFactory()

    @abstractmethod
    def create_button(self) -> Button:
        pass

    @abstractmethod
    def create_menu(self) -> Menu:
        pass

    @staticmethod
    def read_from_config_file(key: str) -> int:
        if key == "OS_TYPE":
            # Assuming 0 represents Windows and 1 represents Linux
            return 1
        return -1
